ci/jenkins/scale-agent.sh

```diff
@@ -0,0 +1,226 @@
+        --label                  Label for the jenkins agent.
+        --kind                   Setup kind testbed."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --workdir)
+    WORKDIR="$2"
+    shift 2
+    ;;
+    --setup-only)
+    RUN_SETUP_ONLY=true
+    shift
+    ;;
+    --cleanup-only)
+    RUN_CLEANUP_ONLY=true
+    shift
+    ;;
+    --kind)
+    KIND=true
+    shift
+    ;;
+    --jenkins-user)
+    JENKINS_USER="$2"
+    shift 2
+    ;;
+    --jenkins-token)
+    JENKINS_TOKEN="$2"
+    shift 2 
+    ;;
+    --label)
+    LABEL="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function check_jobs_in_queue {
+    local LABEL=$1
+    local SUCCESS=0
+
+    for i in {1..2}; do
+        QUEUE_JOBS=$(curl -k -s -u "$JENKINS_USER:$JENKINS_API_TOKEN" "$JENKINS_URL/queue/api/json")
+
+        JOB_FOUND=$(echo "$QUEUE_JOBS" | jq -r '.items[] | select(.task.labels != null) | .task.labels[]' | grep -w "$LABEL")
+
+        if [ -n "$JOB_FOUND" ]; then
+            echo "Job with label $LABEL found in queue. Retrying in 10 seconds..."
+            SUCCESS=1
+            sleep 10
+        else
+            echo "No job with label $LABEL found in queue. Exit."
+            SUCCESS=0
+            break
+        fi
+    done
+
+    echo $SUCCESS
+}
+
+function add_agent {
+    echo "Checking if there are jobs in the Jenkins queue with label: $LABEL"
+    JOB_EXISTS=$(check_jobs_in_queue "$LABEL")
+
+    if [ "$JOB_EXISTS" -ne 0 ]; then
+        echo "Jobs with label $LABEL not found in the queue. Exit."
+        return 1
+    fi
+
+    echo "Jobs with label $LABEL exist in the queue. Proceeding with agent creation..."
+    sudo gcloud compute instances create "$NEW_AGENT_NAME" --zone="$ZONE" --machine-type="$MACHINE_TYPE" --image-family="$IMAGE_FAMILY" --image-project="$IMAGE_PROJECT" --boot-disk-size="$BOOT_DISK_SIZE"
+    if [ $? -ne 0 ]; then
+        echoerr "Failed to create VM instance $NEW_AGENT_NAME."
+        return 1
+    fi
+
+    echo "Waiting for External IP of $NEW_AGENT_NAME to be available..."
+    IP_READY=false
+    for i in {1..18}; do
+        external_ip=$(gcloud compute instances describe "$NEW_AGENT_NAME" --zone="$ZONE" --format='get(networkInterfaces[0].accessConfigs[0].natIP)')
+        
+        if [ -n "$external_ip" ]; then
+            echo "External IP $external_ip is now available."
+            IP_READY=true
+            break
+        else
+            echo "External IP not ready yet. Waiting 10 seconds..."
+            sleep 10
+        fi
+    done
+
+    if [ "$IP_READY" = false ]; then
+        echo "External IP for $NEW_AGENT_NAME did not become ready within 3 minutes. Exiting..."
+        return 1
```

Comment:
It can delete the Jenkins VM if external IP doesn't work. Otherwise the VM is running but it doesn't join Jenkins.

Is the external IP a prerequisite for running gcloud compute ssh?